byterun/caml/misc.h

```diff
@@ -124,6 +124,43 @@ CAMLnoreturn_end;
 CAMLextern char * caml_strdup(const char * s);
 CAMLextern char * caml_strconcat(int n, ...); /* n args of const char * type */
 
+/* Integer arithmetic with overflow detection */ 
```

Comment:
I think it might be a good idea to have a comment here that specifies the sense of the return value.